airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -86,6 +89,24 @@ def _authorize(self):
                                        'use a JSON key file.')
             else:
                 raise AirflowException('Unrecognised extension for key file.')
+        else:
+            if not scope:
+                raise AirflowException('Scope should be defined when using key JSON.')
+            scopes = [s.strip() for s in scope.split(',')]
+
+            # Get credentials from JSON data provided in the UI.
+            try:
+                keyfile_dict = json.loads(keyfile_dict)
+
+                # Depending on how the JSON was formatted, it may contain
+                # escaped newlines. Convert those to actual newlines.
+                keyfile_dict['private_key'] = keyfile_dict['private_key'].replace(
+                    '\\n', '\n')
+
+                credentials = ServiceAccountCredentials \
+                    .from_json_keyfile_dict(keyfile_dict, scopes)
+            except json.decoder.JSONDecodeError:
+                raise AirflowException('Invalid key JSON.')
 
         http = httplib2.Http()
         return credentials.authorize(http)
```

Comment:
Note: In the original commit (https://github.com/apache/airflow/commit/b3e985a3146272ecfd3ceaaa0d8567e4e9e117d4), this was just `return  credentials`.